pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -292,12 +285,48 @@ func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 			s.deletePods(node.Name)
 		}
 	}
-
 	conditions = append(conditions, *newReadyCondition)
 
+	// Check Condition: NodeSchedulable
+	oldSchedulableCondition := s.getCondition(node, api.NodeSchedulable)
+	newSchedulableCondition := s.checkNodeSchedulable(node)
```

Comment:
I think node status should reflect the current status and it may have multiple node conditions in failed state. Subset of the node conditions may impact other components in the system.